### DISCUSSION TITLE
Photos not synchronising from Qfield to Cloud

### DISCUSSION THREAD

**Discussion by Rubendbdv** [opengisch/QField#3983] [Q&A]
Greetings,

We recently had a problem with photos not synchronising on the cloud. For this project we place points (trees) and fill in attributes (information regarding the trees), as well as photos. The attributes get pushed through with synchronising the cloud, but the photos don't. This problem happens with photos from seemingly random times and with 2 types of connections (pushing with WIFI or Mobile Network gives the same result)

Specifics we use:
-Tablet (Trimble R2) with Qfield (version 2.6.3)
-Desktop (different devices, OS Windows 11) with Qgis 3.22.15
-Connection with WIFI aswell as Mobile Network
-Qfieldcloud username: "thomasdbdv"
-Project: "boombeheer_maasmechelen"

Normally we place a point and the attributes and photos get pushed through. We synchronise via Qgis and we can see the attributes AND photos on our desktop.

For the past days (since February 1st) SOME photos are only stored locally on the tablet but are not pushed through with the rest of the attributes. We can still view the photos when we click on the point in Qfield, but not on desktop Qgis.

Now, the bigger issue is that whenever we synchronise the same project in the future using Qgis on desktop, it automatically deletes these photos from our local folder (when we 'prefer cloud'). 

<img width="419" alt="qgis-ltr-bin_Q4LSlIyTuK" src="https://user-images.githubusercontent.com/125654399/219621703-1cce994a-efb5-43be-8680-59c80bc67f3f.png">

So far the only way we could solve this was manually copying the photos from the DCIM folder on the tablet to the folder on our desktop where Qgis normally stores the photos.

We can see this aswell in our libraries: the DCIM folder where the photos are normally stored now has 277 items that are synched by cloud. Now when we copy the items (371) out of the local folder (exported from tablet to desktop) into this cloud folder, the total amount of items becomes 424, where there are 224 items with the same name.

<img width="1152" alt="explorer_BJKqC4VKFT" src="https://user-images.githubusercontent.com/125654399/219632154-f04e9012-3c10-499e-9653-7fc9b5859686.png">

This also shows that the photos that won't synchronise to the cloud are not from a specific timeframe, but rather in groups (the highlighted items are the ones copied from the local exported folder, the non-highlighted items are the ones already synchronised):

<img width="1152" alt="explorer_UuM517rjLa" src="https://user-images.githubusercontent.com/125654399/219637609-ea7a70e9-69d5-434f-accc-bd9c89c48e77.png">
<img width="1152" alt="explorer_C7kvchTtQg" src="https://user-images.githubusercontent.com/125654399/219642342-09688fe1-ddc9-4059-9816-6b480594bec6.png">

The steps we take:

-Place a point in a project using Qfield.
-Fill in information (attributes) and a photo.
-Push changes in the Qfield app.
-Qfield gives notification that the changes have been pushed through, but it doesn't say anything about attachments (before it said: "(amount) attachments are being uploaded in the background"
-Synchronise on desktop using Qfield cloud plugin
-Qgis cloud plugin sees that the cloud has less photos in it than the local folder on our desktop
-Qgis cloud plugin wants to delete them (because we 'prefer cloud')

Even though we can solve this problem manually at the moment, everytime we synchronise in the future it appears again, and we don't know if it will happen with other projects. 

All help is greatly appreciated.

Sincerely,
Ruben

**Comment by ZsanettMed** [comment]
Hi Ruben, 

I assume it is just a misunderstanding. "prefer cloud" means you prefer the status of the cloud. E.g. On the QFielCloud are 10 Photos available and on your desktop 20, after sync, the same appears on your desktop, that means 10 photos. Just look again closely to your screenshot :) 
![image](https://user-images.githubusercontent.com/83091224/221861453-f9719dd3-e305-4e95-8ff6-93cd14e1d663.png)

**Reply by Rubendbdv** [reply]
Greetings,

Thank you for your reply. 

I used this screenshot as an example to show that QfieldCloud indeed has less photos available than my desktop, but this is only the case because I manually exported the DCIM folder from the tablet to my desktop. I manually exported these photos because we use a layout in our reports where a photo is placed. This photo has a path to my local cloud folder on my desktop (the one where photos get deleted if I prefer cloud status). 


We still work with this app every day, and all the attributes except for the photos get pushed through when I press 'push changes'. This means I have to manually export the DCIM folder for every report I make.

In short: the synchronising function works fine, it does exactly what it's supposed to do, only the 'push' function on my tablet doesn't push the photos to the cloud.

Sincerely,
Ruben

**Reply by ZsanettMed** [reply]
Hi Ruben, 
Apologize! The issue is confirmed. Thank you very much for reporting and nice to hear, that you found a workaround. We will inform you as soon as the problem is solved. 
Best regards, 
Zsanett

**Reply by serverul** [reply]
The issue still persists on iOS. Any changes soon?

**Reply by vk3il** [reply]
I am having the same issue on Android. I also can't find the files on my Android device to manually copy, so this it becoming a real problem for me. Is there any progress on a resolution?

I can't see this in the Issues log, so if there's no response shortly, I will create this as an issue.

Thanks!

**Reply by serverul** [reply]
It is also an issue regardin how you arrange the project, as I saw. 
You can see below how I set it up and it works flawlessly since then on android. Still the issue persists on iOS.
![qgis_photo](https://github.com/opengisch/QField/assets/126231162/28b49eb1-b42c-48c1-9f05-c1c484b94230)

**Comment by NVERRIER89** [comment]
We also have the same sync issue regarding photos.

Some projects are working very well. For other projects, we are unable to recover the photos despite several synchronization attempts.

It becomes urgent to address the issue as we use the app every day.

Name of the organization: **OPTIQUE-TELECOMS**

Thank you and good luck.

**Reply by serverul** [reply]
Check my reply above to see if it suits your issue. 
Also you can contact me directly if you need further assistance.

**Comment by vk3il** [comment]
@serverul Thanks for your suggestion. I've adjusted my configuration to be identical to the one you've provided. Unfortunately it is still not synchronising the photos (android).

**Comment by georgejr809** [comment]
I'm working at a nature-based solutions company, and we're using QField Cloud as well, and this is also my problem. I did the setup as shown in the attached screenshot, but some of the images still do not sync, and I don’t know why. The problem is that we’re covering a very large area here in the Philippines, and the people using the tablets for QField are not very inclined to technology, which is why we can’t manually extract the photos from their tablets. Going to those forest guards and manually extracting their photos is very expensive, and that’s why we purchased QField Cloud, but the problem is still the same. Hopefully this will be resolved.
![qfield](https://github.com/user-attachments/assets/3cc92e29-4235-4630-8333-b20b91355542)

**Reply by nirvn** [reply]
@georgejr809 , I see you've properly identified the method through which you can manually push images to QFieldCloud. Did that help you recover the missing photos or there are still missing ones even _after_ manually pushing the DCIM folder content to the cloud?

**Reply by nirvn** [reply]
@georgejr809 , is it also possible you are using one login credential across multiple devices? If so, that would create some interference.

**Reply by georgejr809** [reply]
@nirvn  thanks for the follow-up.

Yes, I used the “Push to QFieldCloud” option on the DCIM folder on the devices. That did bring in some photos, but there are still photos missing even after the manual push. To get all the photos I still have to manually export / copy them from the devices, but that defeats the purpose of using the cloud sync.

**Reply by nirvn** [reply]
@georgejr809 , indeed, manually copying does defeat the purpose  :)  FYI, we have made significant improvements on that front for the next version of QField (released mid Dec.), I'm hoping this will fix your problem. 

To follow up on the one unanswered question from above: are you sharing a login across multiple devices?

**Reply by georgejr809** [reply]
@nirvn thanks for the follow-up.
Yes — in our workflow some forest guards do use the same QFieldCloud login across multiple devices. This is needed because when they are in the field (often deep in the forest) they can spend up to 3 days out there, and a tablet can run low on battery or fail.
To avoid losing work, we carry spare devices, and they sometimes transfer/log in with the same account on another device to continue data collection.

**Reply by nirvn** [reply]
@georgejr809 , OK, I suspect this is partly why you're seeing this. With that detail confirmed, we'll look into it some more.
